Data/ByteString/Short/Internal.hs

```diff
@@ -524,12 +526,18 @@ copyByteArray (BA# src#) (I# src_off#) (MBA# dst#) (I# dst_off#) (I# len#) =
 ------------------------------------------------------------------------
 -- FFI imports
 
-memcmp_ByteArray :: BA -> BA -> Int -> IO CInt
+memcmp_ByteArray :: BA -> BA -> Int -> Int
+#if MIN_VERSION_base(4,11,0)
+memcmp_ByteArray (BA# ba1#) (BA# ba2#) (I# len#) =
+  I# (compareByteArrays#  ba1# 0# ba2# 0# len#)
+#else
 memcmp_ByteArray (BA# ba1#) (BA# ba2#) len =
```

Comment:
I'm slightly inclined to adopt the "upstream naming", but it doesn't matter very much to me.

```suggestion
compareByteArrays :: BA -> BA -> Int -> Int
#if MIN_VERSION_base(4,11,0)
compareByteArrays (BA# ba1#) (BA# ba2#) (I# len#) =
  I# (compareByteArrays#  ba1# 0# ba2# 0# len#)
#else
compareByteArrays (BA# ba1#) (BA# ba2#) len =
```